jgrapht-core/src/main/java/org/jgrapht/alg/matching/GreedyWeightedMatching.java

```diff
@@ -92,18 +92,22 @@ public GreedyWeightedMatching(Graph<V, E> graph, double epsilon)
      * @return a matching
      */
     @Override
-    public Matching<E> getMatching()
+    public Matching<V, E> getMatching()
     {
         // sort edges in non-decreasing order of weight
         // (the lambda uses e1 and e2 in the reverse order on purpose)
         List<E> allEdges = new ArrayList<>(graph.edgeSet());
         Collections.sort(
-            allEdges,
-            (e1, e2) -> comparator.compare(graph.getEdgeWeight(e2), graph.getEdgeWeight(e1)));
+                allEdges,
```

Comment:
Not sure about this one. 

Does this change still guarantee a 2-approximation? If not, I would create another class which would (perhaps two subclasses one with each comparator) clearly state its properties and have a different name. Maybe have a flag which allows the user to switch between the two and clearly document that in the javadocs?